crates/sequencer/src/state.rs

```diff
@@ -16,9 +22,235 @@ use starknet_api::{
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_storage() {
+        // Given
+        let mut state = &mut State::default();
+
+        // When
+        state.set_storage_at(*TEST_CONTRACT_ADDRESS, StorageKey(*ONE_PATRICIA), *ONE_FELT);
+
+        // Then
+        let expected = *ONE_FELT;
+        let actual = state
+            .get_storage_at(*TEST_CONTRACT_ADDRESS, StorageKey(*ONE_PATRICIA))
+            .unwrap();
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_nonce() {
+        // Given
+        let mut state = &mut State::default();
+
+        // When
+        state.increment_nonce(*TEST_CONTRACT_ADDRESS).unwrap();
+
+        // Then
+        let expected = Nonce(*ONE_FELT);
+        let actual = state.get_nonce_at(*TEST_CONTRACT_ADDRESS).unwrap();
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_class_hash() {
+        // Given
+        let mut state = &mut State::default();
+
+        // When
+        state
+            .set_class_hash_at(*TEST_CONTRACT_ADDRESS, *ONE_HASH)
+            .unwrap();
+
+        // Then
+        let expected = *ONE_HASH;
+        let actual = state.get_class_hash_at(*TEST_CONTRACT_ADDRESS).unwrap();
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_contract_class() {
+        // Given
+        let mut state = &mut State::default();
+
+        // When
+        state
+            .set_contract_class(&ONE_HASH, ContractClass::V0(ContractClassV0::default()))
+            .unwrap();
+
+        // Then
+        let expected = ContractClass::V0(ContractClassV0::default());
+        let actual = state.get_compiled_contract_class(&ONE_HASH).unwrap();
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "UndeclaredClassHash(ClassHash(StarkFelt(\"0x0000000000000000000000000000000000000000000000000000000000000001\")))"
+    )]
+    fn test_uninitialized_contract_class() {
+        // Given
+        let mut state = &mut State::default();
+
+        // When
+        state.get_compiled_contract_class(&ONE_HASH).unwrap();
+    }
+
+    #[test]
+    fn test_compiled_class_hash() {
+        // Given
+        let mut state = &mut State::default();
+
+        // When
+        state
+            .set_compiled_class_hash(*ONE_HASH, *ONE_COMPILED_HASH)
+            .unwrap();
+
+        // Then
+        let expected = *ONE_COMPILED_HASH;
+        let actual = state.get_compiled_class_hash(*ONE_HASH).unwrap();
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "UndeclaredClassHash(ClassHash(StarkFelt(\"0x0000000000000000000000000000000000000000000000000000000000000001\"))"
+    )]
+    fn test_uninitialized_compiled_class_hash() {
+        // Given
+        let mut state = &mut State::default();
+
+        // When
+        state.get_compiled_class_hash(*ONE_HASH).unwrap();
```

Comment:
isn't it better to not unwrap and assert on the error? just an idea of preference

Reply:
This was something that was discussed with @ClementWalter, if I recall correctly he linked a part of the rust book where it indicated you should use #[should_panic]

Reply:
It's here, reading through it I am not sure, we could do `is_err()` and then also assert on the specific error with `matches` or something similiar.
https://doc.rust-lang.org/book/ch11-01-writing-tests.html?highlight=should_panic#checking-for-panics-with-should_panic